lib/opentelemetry_plug.ex

```diff
@@ -100,42 +99,34 @@ defmodule OpentelemetryPlug do
 
   @doc false
   def handle_stop(_, _measurements, %{conn: conn}, _config) do
-    if in_span?() do
-      Tracer.set_attribute("http.status_code", conn.status)
-      # For HTTP status codes in the 4xx and 5xx ranges, as well as any other
-      # code the client failed to interpret, status MUST be set to Error.
-      #
-      # Don't set the span status description if the reason can be inferred from
-      # http.status_code.
-      if conn.status >= 400 do
-        Tracer.set_status(OpenTelemetry.status(:error, ""))
-      end
-
-      Tracer.end_span()
-      restore_parent_ctx()
+    Tracer.set_attribute(:"http.status_code", conn.status)
+    # For HTTP status codes in the 4xx and 5xx ranges, as well as any other
+    # code the client failed to interpret, status MUST be set to Error.
+    #
+    # Don't set the span status description if the reason can be inferred from
+    # http.status_code.
+    if conn.status >= 400 do
+      Tracer.set_status(OpenTelemetry.status(:error, ""))
     end
+
+    Tracer.end_span()
   end
 
   @doc false
   def handle_exception(_, _measurements, metadata, _config) do
-    if in_span?() do
-      %{kind: kind, reason: reason, stacktrace: stacktrace} = metadata
-      exception = Exception.normalize(kind, reason, stacktrace)
-
-      Span.record_exception(
-        Tracer.current_span_ctx(),
-        exception,
-        stacktrace
-      )
-
-      Tracer.set_status(OpenTelemetry.status(:error, Exception.message(exception)))
-      Tracer.set_attribute("http.status_code", 500)
-      Tracer.end_span()
-      restore_parent_ctx()
-    end
-  end
+    %{kind: kind, reason: reason, stacktrace: stacktrace} = metadata
+    exception = Exception.normalize(kind, reason, stacktrace)
+
+    Span.record_exception(
+      Tracer.current_span_ctx(),
+      exception,
+      stacktrace
+    )
 
-  defp in_span?, do: Tracer.current_span_ctx() != :undefined
+    Tracer.set_status(OpenTelemetry.status(:error, Exception.message(exception)))
+    Tracer.set_attribute(:"http.status_code", 500)
+    Tracer.end_span()
```

Comment:
The span will be ended and the status code set in the stop handler. These are safe to delete. :)

Reply:
The `stop` handler is _not_ called in the exception case. See https://hexdocs.pm/elixir/Kernel.SpecialForms.html#try/1-else-clauses

> Only the result of the tried expression falls down to the `else` clause. If the `try` ends up in the `rescue` or `catch` clauses, their result will not fall down to `else`

Reply:
Ok. This is being handled differently than in Phoenix then. Phoenix will catch the error again and complete the plug pipeline back out.